rcgen/src/lib.rs

```diff
@@ -397,17 +418,14 @@ impl DistinguishedName {
 Iterator over [`DistinguishedName`] entries
 */
 pub struct DistinguishedNameIterator<'a> {
-	distinguished_name: &'a DistinguishedName,
-	iter: std::slice::Iter<'a, DnType>,
+	iter: std::slice::Iter<'a, (DnType, DnValue)>,
 }
 
 impl<'a> Iterator for DistinguishedNameIterator<'a> {
 	type Item = (&'a DnType, &'a DnValue);
 
 	fn next(&mut self) -> Option<Self::Item> {
-		self.iter
-			.next()
-			.and_then(|ty| self.distinguished_name.entries.get(ty).map(|v| (ty, v)))
+		self.iter.next().map(|(key, value)| (key, value))
```

Comment:
Looks like the `map()` call is unnecessary?

Reply:
It's necessary to map from `&(key, val)` to `(&key, &val)`:

![image](https://github.com/user-attachments/assets/9dbdeb0c-6432-4d4e-bca3-e0d768636304)

If there is a more ergonomic way to do this let me know.